qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDiagnostics.java

```diff
@@ -541,7 +544,11 @@ private ResolvedJavaTypeInfo validateObjectPart(String namespace, ObjectPart obj
 
 			// ex : {item} --> undefined variable
 			Range range = QutePositionUtility.createRange(objectPart);
-			Diagnostic diagnostic = createDiagnostic(range, DiagnosticSeverity.Warning, QuteErrorCode.UndefinedVariable,
+			DiagnosticSeverity severity = validationSettings.getUndefinedObject().getDiagnosticSeverity();
+			if (severity == null) {
+				severity = DiagnosticSeverity.Warning;

```

Comment:
`return null;`

Reply:
The range should be created only if severity is not null. So move Range range = QutePositionUtility.createRange(objectPart); after if (severity == null) return null;